src/test/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantSourceTest.java

```diff
@@ -0,0 +1,165 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.testing.DataflowAssert;
+import com.google.cloud.dataflow.sdk.testing.TestPipeline;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.common.collect.Lists;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFCodec;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.utils.variant.SkeletonVariant;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+import org.broadinstitute.hellbender.utils.variant.VariantContextVariantAdapter;
+import org.broadinstitute.hellbender.utils.variant.VariantUtils;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.broadinstitute.hellbender.utils.variant.VariantContextVariantAdapterTest.createVariantContextVariantAdapterForTesting;
+
+public class VariantSourceTest extends BaseTest {
+    private static final String FEATURE_DATA_SOURCE_TEST_DIRECTORY = publicTestDir + "org/broadinstitute/hellbender/engine/";
+    private static final File QUERY_TEST_VCF = new File(FEATURE_DATA_SOURCE_TEST_DIRECTORY + "feature_data_source_test.vcf");
+
+    static UUID defaultUUID() {
+        return new UUID(0L, 0L);
+    }
+
+    static class clearUUIDDoFn extends DoFn<Variant, Variant> {
+        private static final long serialVersionUID = 1L;
+        @Override
+        public void processElement(ProcessContext c) throws Exception {
+            VariantContextVariantAdapter variantContextVariantAdapterForTesting =
+                    createVariantContextVariantAdapterForTesting(((VariantContextVariantAdapter) c.element()), defaultUUID());
+            c.output(variantContextVariantAdapterForTesting);
+        }
+    }
+
+    @Test(dataProvider = "RealVariantData")
+    public void testVariantPCollection(final List<Variant> variants) {
+        for (int i = 0; i < variants.size(); ++i) {
+            // Update the element with a cleared UUID.
+            variants.set(i, createVariantContextVariantAdapterForTesting(((VariantContextVariantAdapter) variants.get(i)), defaultUUID()));
+        }
+        // Now make a PCollection, to verify that variants can be coded.
```

Comment:
Does it really check that? I asked them to, but last I checked it just made sure that a coder was assigned, not that the coder actually worked.